libshaderc/include/shaderc.h

```diff
@@ -31,6 +31,12 @@ typedef enum {
   shaderc_glsl_tess_evaluation_shader
 } shaderc_shader_kind;
 
+typedef enum {
+  shaderc_msg_default           = 0x00,
+  shaderc_msg_spirv_rules       = 0x01,
```

Comment:
This presents the wrong abstraction.
What you have for "shaderc_msg_spirv_rules" is "generate SPIR-V using OpenGL conventions".   The default has always been to target OpenGL conventions, but glslang needed the extra enum to control whether generate SPIR-V.

See in glslang's Standalone.cpp, function SetMessageOptions:
https://github.com/KhronosGroup/glslang/blob/master/StandAlone/StandAlone.cpp#L601
It says "Translate the meaningful subset of command-line options to parser-behaviour options."

Look further inside glslang and you see that it controls what set of builtin functions are available.  

So it's really about selecting dialect depending on target environment: either OpenGL or Vulkan. These are mutually exclusive.

So I'd rather have something like this:

```
// Selects which target environment conventions to follow in generated SPIR-V.
// Each target environment such as Vulkan or OpenGL should define the conventions
// expected of a SPIR-V module.
enum {
shaderc_target_env_vulkan
shaderc_target_env_opengl,
shaderc_target_env_default = shader_target_env_vulkan
} shaderc_target_env;
```

Eventually this would also be exposed on the glslc command line as --target-env=<something> where that <something> is either vulkan or opengl.  (Default is Vulkan.)